### PR TITLE
Documentation for explain prediction

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: doc/widgets/explain-pred.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+**Outputs**
+
+- Scores: The SHAP value of each features value. Features that contribute more to prediction have higher score deviation from the 0.
+
+**Explain Prediction** widget explains classification or regression model's prediction for the provided data instance. The widget shows what features affect the prediction of selected class the most and how they contribute (towards or against the prediction). The explanation is computed with removing features, replacing them with different options from the background data, and observing the change in the prediction.
+![](images/Explain-Prediction.png)
+
+1. Select the target class -- the plot will show explanations for this class.
+2. Zoom in/out the plot.
+3. Observe the prediction probability for a class and base vale -- an average probability in the dataset.
```
Comment (ajdapretnar) [inline doc/widgets/explain-pred.md:21]: base vale --> base value?

File: doc/widgets/explain-pred.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+4. Plot which shows features that affect the prediction the most (features with longer tape length) and how they affect it. Red features increase the probability for a selected class while blue features decrease the probability. On the right from the tape, you can see the feature name and its value\* for the selected instance. The length of the tape segment (and number on the tape) represent the SHAP value for feature contribution -- it is how much feature the feature affects the probability for the selected class.
```
Comment (ajdapretnar) [inline doc/widgets/explain-pred.md:22]: it is how much feature the feature --> it is how much the feature?

File: doc/widgets/explain-pred.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+6. Get help, save the plot, make the report, or observe the size of input and output data.
+
+\* Some models (including logistic regression) extend the categorical feature to more features with the technique named [one-hot encoding](https://en.wikipedia.org/wiki/One-hot). It means each value in the feature gets a new column which has value 0 (the instance have no this feature value) or 1 (the instance have this feature value) for each instance. In those cases categorical features will be labeled with the format `feature-name=feature-value = 0/1` -- e.g. `chest pain=asymptomatic = 1`. It means that the feature chest pain has value asymptomatic. Model, in this case, made more columns for feature chest pain, one of them was asymptomatic, and it was the case for the selected data instance.
```
Comment (ajdapretnar) [inline doc/widgets/explain-pred.md:25]: the instance have no this feature value --> the instance does not have this feature value

Comment (ajdapretnar) [reply doc/widgets/explain-pred.md:25]: the instance have this feature value --> the instance has this feature value

File: doc/widgets/explain-pred.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Example
+-------
+ 
+First, we open Hearth disease dataset in File widget. With the Data Sampler widget, we split the dataset on the training and test set. The training set is used to train the logistic regression model with the Logistic Regression widget. We compute predictions for the test part of the dataset (remaining data from Data Sampler) with the Predictions widget. In the Predictions widget (the left window) we select the data instance whose prediction we would like to explain -- we select the third row in the data which is predicted to belong to class 1 (diagnosed heart disease).
```
Comment (ajdapretnar) [inline doc/widgets/explain-pred.md:30]: Hearth disease --> heart disease

File: doc/widgets/explain-pred.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+![](images/Explain-Prediction.png)
+
+1. Select the target class -- the plot will show explanations for this class.
+2. Zoom in/out the plot.
+3. Observe the prediction probability for a class and base value -- an average probability in the dataset.
+4. Plot which shows features that affect the prediction the most (features with longer tape length) and how they affect it. Red features increase the probability for a selected class while blue features decrease the probability. On the right from the tape, you can see the feature name and its value\* for the selected instance. The length of the tape segment (and number on the tape) represent the SHAP value for feature contribution -- it is how much the feature affects the probability for the selected class. Edino mogoče bi še dodala, kaj sta ti dve sivi številki na levi. 0.96 in 0.45 na primeru iz dokumentacije. Numbers in the gray boxes indicate the prediction probability for the selected class is (0.6) and the baseline probability (0.45) (the average probability in the data).
```
Comment (ajdapretnar) [inline doc/widgets/explain-pred.md:22]: > Edino mogoče bi še dodala, kaj sta ti dve sivi številki na levi. 0.96 in 0.45 na primeru iz dokumentacije.

😆

Comment (PrimozGodec) [reply doc/widgets/explain-pred.md]: Haha. I accidentally pasted wrong text and then it seems removed just part of it. Now it should be ok

**Pull request merged**